processor/spanmetricsprocessor/cache.go

```diff
@@ -0,0 +1,57 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanmetricsprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/spanmetricsprocessor"
+
+import lru "github.com/hashicorp/golang-lru"
+
+// Cache is consist of an LRU cache and the evicted items from the LRU cache
+// this data structure makes sure all the cached item can be retrieved either from the LRU cache or the evictedItems map
+// In spanmetricsprocessor's use case, we need to hold all the items during the current processing step for building the
+// metrics. The evicted items can/should be safely removed once the metrics are built from the current batch of spans.
```

Comment:
Please end sentences with full-stops.
```suggestion
// Cache is consist of an LRU cache and the evicted items from the LRU cache.
// This data structure makes sure all the cached items can be retrieved either from the LRU cache or the evictedItems map.
// In spanmetricsprocessor's use case, we need to hold all the items during the current processing step for building the
// metrics. The evicted items can/should be safely removed once the metrics are built from the current batch of spans.
```